provider/openstack/provider.go

```diff
@@ -719,9 +724,25 @@ func newCredentials(spec environs.CloudSpec) (identity.Credentials, identity.Aut
 		cred.ProjectDomain = credAttrs[CredAttrProjectDomainName]
 		cred.UserDomain = credAttrs[CredAttrUserDomainName]
 		cred.Domain = credAttrs[CredAttrDomainName]
-		authMode = identity.AuthUserPass
-		if cred.Domain != "" || cred.UserDomain != "" || cred.ProjectDomain != "" {
+		if credAttrs[CredAttrVersion] != "" {
```

Comment:
Based on the last comment, we want credAttrs[CredAttrVersion] to be "" if the identity version is not specified